### PR TITLE
added documentation for Dates.TimeZone, Dates.UTC

### PR DESCRIPTION
**Pull request by mrdofu**
see #31202

### CODE REVIEW

File: stdlib/Dates/src/types.jl [hunk 1]
```diff
@@ -83,7 +83,20 @@ abstract type Calendar <: AbstractTime end
 # ISOCalendar provides interpretation rules for UTInstants to civil date and time parts
 struct ISOCalendar <: Calendar end
 
+"""
+    TimeZone
+
+Geographic zone generally based on longitude determining what the time is at a certain location.
+Some time zones observe daylight savings (eg EST -> EDT).
```
Comment (quinnj) [inline stdlib/Dates/src/types.jl:90]: I think it'd be helpful to add a mention of the [TimeZones.jl](https://github.com/JuliaTime/TimeZones.jl) package here; if people are looking at `TimeZone` docs, they most likely need the package anyway.

File: stdlib/Dates/src/types.jl [hunk 2]
```diff
@@ -85,7 +85,21 @@ abstract type Calendar <: AbstractTime end
...
+For implementations and more support, see the `TimeZones` package
```
Comment (quinnj) [inline stdlib/Dates/src/types.jl:93]:
```suggestion
For implementations and more support, see the [`TimeZones.jl`](https://github.com/JuliaTime/TimeZones.jl) package
```

**Comment by KristofferC** [comment]
Perhaps better to squash this at merge?

**Comment by quinnj** [comment]
Oops; I guess I missed that option. I had committed my suggestion and I don't remember seeing the option to squash or not; but I probably just missed it.

**Pull request merged**